framework/src/executioners/Transient.C

```diff
@@ -236,23 +266,37 @@ Transient::execute()
     _problem.advanceState();
 
   // Start time loop...
+  bool first = true; // needs to not be restartable data
```

Comment:
We've messed with this several times, it's definitely a hack and used by some materials in tensor mechanics among other things. I don't think it's as easy to make go away as you might think.